shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1332,6 +1343,15 @@ - (void)handleKeyboardNotification:(NSNotification*)notification {
 
   self.targetViewInsetBottom = calculatedInset;
   NSTimeInterval duration = [info[UIKeyboardAnimationDurationUserInfoKey] doubleValue];
+
+  BOOL keyboardWillShow = beginKeyboardFrame.origin.y > keyboardFrame.origin.y;
+
+  // Flag for simultaneous compounding animation calls.
+  self.keyboardAnimationIsCompounding = self.keyboardAnimationIsShowing == keyboardWillShow;
+
+  // Mark keyboard as showing or hiding.
+  self.keyboardAnimationIsShowing = keyboardWillShow;
+
   [self startKeyBoardAnimation:duration];
```

Comment:
- Maybe better to add more comments to explain when this `compounding` will happen. ^_^

- I think we can use 
``` objc
BOOL keyboardAnimationIsCompounding = self.keyboardAnimationIsShowing == keyboardWillShow
 && _keyboardAnimationVSyncClient != nil;
```
 which seems better to indicate current animation is running and the direction is the same (Because the `_keyboardAnimationVSyncClient` is the animator, which is more readable)

so we can use this to avoid double triggering animation:
```objc
if (!keyboardAnimationIsCompounding) {
    [self startKeyBoardAnimation:duration];
}
```
So we can make `keyboardAnimationIsCompounding` to a temporary variable and **remove below code**

``` objc
else if (self.keyboardAnimationIsCompounding) {
    // If keyboardAnimationView has superview, means current animation is running,
    // and if keyboard animation is in the same direction as current animation, ignore it.
    return;
}
```

- Maybe better to add a new test for this compounding animation scenario O(∩_∩)O.

Reply:
👍🏻 updated this and added a new test